### PR TITLE
indexer: Perform Bedrock DB upgrade

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

**Comment by mslipper** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3665** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3665" target="_blank">Graphite</a> 
    * **PR #3666** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3666" target="_blank">Graphite</a>  👈
      * **PR #3667** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3667" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3666?utm_source=stack-comment).

File: indexer/db/sql.go [hunk 1]
```diff
@@ -125,8 +125,8 @@ CREATE TABLE IF NOT EXISTS airdrops (
 const updateWithdrawalsTable = `
 ALTER TABLE withdrawals ADD COLUMN IF NOT EXISTS br_withdrawal_hash VARCHAR NULL;
 ALTER TABLE withdrawals ADD COLUMN IF NOT EXISTS br_withdrawal_finalized_tx_hash VARCHAR NULL;
-ALTER TABLE withdrawals ADD COLUMN IF NOT EXISTS br_withdrawal_finalized_log_index BOOLEAN NULL;
```
Comment (roninjin10) [inline indexer/db/sql.go:128]: The name and the type here used to be very confusing nice change

**Review by roninjin10 — APPROVED** [review]

File: indexer/db/db.go [hunk 2]
```diff
@@ -209,26 +215,39 @@ func (d *Database) AddIndexedL1Block(block *IndexedL1Block) error {
...
+					deposit.ToAddress.String(),
+					deposit.L1Token.String(),
+					deposit.L2Token.String(),
+					deposit.Amount.String(),
+					deposit.TxHash.String(),
+					deposit.LogIndex,
+					block.Hash.String(),
+					deposit.Data,
+				)
+				if err != nil {
```
Comment (tynes) [inline indexer/db/db.go:233]: Just want to be sure that that an error wont result in really bad things, ie skipped indexing of deposits or withdrawals. Should we make a db transaction and then commit it at the end?

Comment (mslipper) [reply indexer/db/db.go:233]: It's safe against what you're describing :smile:.

The `txn` wrapper on line 206 wraps all this logic in a DB transaction. Any error in here will cause a rollback. Blocks are indexed one by one, so if the TX is rolled back the indexer will try to index the block again.

**Review by tynes — APPROVED** [review]

**Pull request merged**